### PR TITLE
Add a hook to JSON stringify that makes it only output own properties.

### PR DESCRIPTION
**Pull request by mcarcaso**
I noticed stringify outputting values from expressions and that didn't seem right to me. How do you feel about this change? I could change the default value to preserve the current behaviour but the current behaviour feels wrong to me.

### CODE REVIEW

File: src/foam/core/JSON.js [hunk 1]
```diff
@@ -155,6 +155,11 @@ foam.CLASS({
     },
     {
       class: 'Boolean',
+      name: 'outputOwnPropertiesOnly',
+      value: true
```
Comment (kgrgreer) [inline src/foam/core/JSON.js:159]: This is probably the wrong default value since it changes the behaviour for others.

**Comment by mcarcaso** [comment]
You're right that it changes the behavior but I don't think that behavior
is worth preserving. It seems that the special case would be to output
expressions as I don't know why you'd want that behavior normally.

**Comment by mdittmer** [comment]
Is "own" here "belonging to to concrete class", "set on instance_", or
something else entirely?

**Comment by mcarcaso** [comment]
It's the result of hasOwnProperty which I think is just it being set on
instance_

**Pull request merged**